stream/sentinel-intel/src/sentinel_intel_connector/api_handler.py

```diff
@@ -0,0 +1,236 @@
+        try:
+            url = f"https://login.microsoftonline.com/{self.config.tenant_id}/oauth2/v2.0/token"
+            body = {
+                "client_id": self.config.client_id,
+                "client_secret": self.config.client_secret,
+                "grant_type": "client_credentials",
+                "scope": "https://graph.microsoft.com/.default",
+            }
+            response = requests.post(url, data=body)
+            response_json = response.json()
+            oauth_token = response_json["access_token"]
+
+            self.headers = {"Authorization": oauth_token}
+        except Exception as e:
+            raise ValueError("[ERROR] Failed generating oauth token {" + str(e) + "}")
+
+    def _build_request_body(self, observable: dict) -> dict:
+        """
+        Build Sentinel POST/PATCH request's body from an observable.
+        :param observable: Observable to build body from
+        :return: Dict containing keys/values required for POST/PATCH requests on Sentinel.
+        """
+        body = {
+            "threatType": get_threat_type(observable),
+            "description": get_description(observable),
+            "tags": get_tags(observable),
+            "confidence": observable.get("confidence", 50),
+            "externalId": OpenCTIConnectorHelper.get_attribute_in_extension(
+                "id", observable
+            ),
+            "lastReportedDateTime": OpenCTIConnectorHelper.get_attribute_in_extension(
+                "updated_at", observable
+            ),
+            "expirationDateTime": get_expiration_datetime(
+                observable, int(self.config.expire_time)
+            ),
+            "action": self.config.action or get_action(observable),
+            "tlpLevel": self.config.tlp_level or get_tlp_level(observable),
+            "passiveOnly": "true" if self.config.passive_only else "false",
+            "targetProduct": self.config.target_product,
+        }
+
+        network_types = ["ipv4-addr", "ipv6-addr", "domain-name", "url"]
+        if observable["type"] in network_types:
+            ioc_type = get_ioc_type(observable)
+            body[ioc_type] = observable.get("value", None)
+        elif observable["type"] == "email-addr":
+            body["emailSenderAddress"] = observable.get("value", None)
+            body["emailSenderName"] = observable.get("display_name", None)
+        elif observable["type"] == "file":
+            body["fileHashType"] = get_hash_type(observable)
+            body["fileHashValue"] = get_hash_value(observable)
+            body["fileName"] = observable.get("name", None)
+            body["fileSize"] = observable.get("size", 0)
+            body["fileCreatedDateTime"] = (
+                OpenCTIConnectorHelper.get_attribute_in_extension(
+                    "created_at", observable
+                )
+            )
+        else:
+            body = {}
+        return body
+
+    def get_indicators(self) -> list[dict] | None:
```

Comment:
I found similarities in your code calling the API. You could have a method to reduce code duplication

```
def _api_request(self, method: str, url: str, **kwargs):
    try:
        response = self.session.request(method, url, **kwargs)
        response.raise_for_status()
          self.helper.connector_logger.info(
                f"[API] HTTP {method} Request to endpoint",
                {"url_path": url},
            )
        return response.json()
    except requests.RequestException as err:
        self.helper.connector_logger.error(
            "[API] Error while requesting : ",
                {"url_path": f"{method} {url}", "error": str(err)},
        )
        return None


```